lib/gst/player/gstplayer-media-info.c

```diff
@@ -0,0 +1,623 @@
+gst_player_media_info_class_init (GstPlayerMediaInfoClass * klass)
+{
+  GObjectClass  *oclass = (GObjectClass *) klass;
+
+  oclass->finalize = gst_player_media_info_finalize;
+}
+
+static GstPlayerVideoInfo*
+gst_player_video_info_new (void)
+{
+  return g_object_new (GST_TYPE_PLAYER_VIDEO_INFO, NULL);
+}
+
+static GstPlayerAudioInfo*
+gst_player_audio_info_new (void)
+{
+  return g_object_new (GST_TYPE_PLAYER_AUDIO_INFO, NULL);
+}
+
+static GstPlayerSubtitleInfo*
+gst_player_subtitle_info_new (void)
+{
+  return g_object_new (GST_TYPE_PLAYER_SUBTITLE_INFO, NULL);
+}
+
+static GstPlayerStreamInfo*
+gst_player_video_info_copy (GstPlayerVideoInfo *ref)
+{
+  GstPlayerVideoInfo *ret;
+
+  ret = gst_player_video_info_new ();
+
+  ret->width = ref->width;
+  ret->height  = ref->height;
+  ret->framerate_num = ref->framerate_num;
+  ret->framerate_denom = ref->framerate_denom;
+  ret->par_num = ref->par_num;
+  ret->par_denom = ref->par_denom;
+  ret->bitrate = ref->bitrate;
+  ret->max_bitrate = ref->max_bitrate;
+
+  return (GstPlayerStreamInfo*) ret;
+}
+
+static GstPlayerStreamInfo*
+gst_player_audio_info_copy (GstPlayerAudioInfo *ref)
+{
+  GstPlayerAudioInfo *ret;
+
+  ret = gst_player_audio_info_new ();
+
+  ret->sample_rate = ref->sample_rate;
+  ret->channels = ref->channels;
+  ret->bitrate = ref->bitrate;
+  ret->max_bitrate = ref->max_bitrate;
+
+  if (ref->language)
+    ret->language = g_strdup (ref->language);
+
+  return (GstPlayerStreamInfo*) ret;
+}
+
+static GstPlayerStreamInfo*
+gst_player_subtitle_info_copy (GstPlayerSubtitleInfo *ref)
+{
+  GstPlayerSubtitleInfo *ret;
+
+  ret = gst_player_subtitle_info_new ();
+  if (ref->language)
+    ret->language = g_strdup (ref->language);
+
+  return (GstPlayerStreamInfo*) ret;
+}
+
+GstPlayerStreamInfo*
+gst_player_stream_info_copy (GstPlayerStreamInfo *ref)
+{
+  GstPlayerStreamInfo *info = NULL;
+
+  if (!ref)
+    return NULL;
+
+  if (GST_IS_PLAYER_VIDEO_INFO(ref))
+    info = gst_player_video_info_copy ((GstPlayerVideoInfo*)ref);
+  else if (GST_IS_PLAYER_AUDIO_INFO(ref))
+    info = gst_player_audio_info_copy ((GstPlayerAudioInfo*)ref);
+  else
+    info = gst_player_subtitle_info_copy ((GstPlayerSubtitleInfo*)ref);
+
+  info->stream_index = ref->stream_index;
+  if (ref->tags)
+    info->tags = gst_tag_list_copy (ref->tags);
+  if (ref->caps)
+    info->caps = gst_caps_copy (ref->caps);
+
+  return info;
+}
+
+GstPlayerMediaInfo*
+gst_player_media_info_copy (GstPlayerMediaInfo* ref)
+{
+  GList *l;
+  GstPlayerMediaInfo *info;
+
+  if (!ref)
+    return NULL;
+
+  info = gst_player_media_info_new (ref->uri);
+  info->duration = ref->duration;
+
+  for (l = ref->stream_list; l != NULL; l = l->next) {
+    GstPlayerStreamInfo *s;
+
+    s = gst_player_stream_info_copy((GstPlayerStreamInfo*)l->data);
+    info->stream_list = g_list_append (info->stream_list, s);
+
+    if (GST_IS_PLAYER_AUDIO_INFO(s))
+      info->audio_cachelist = g_list_append (info->audio_cachelist, s);
```

Comment:
This is not really a "cache", maybe just call it audio_stream_list (same for the other two)